ArduPlane/quadplane.cpp

```diff
@@ -2895,6 +2921,39 @@ void QuadPlane::vtol_position_controller(void)
     }
 }
 
+void QuadPlane::assign_tilt_to_fwd_thr(void) {
+#if QAUTOTUNE_ENABLED
+    if (plane.control_mode == &plane.mode_qautotune) {
+        // limiting forward pitch and using forward throttle or rotor tilt is incompatible with auto-tune
+        q_fwd_throttle = 0.0f;
+        q_fwd_nav_pitch_lim_cd = -aparm.angle_max;
+        return;
+    }
+#endif
+    if (is_positive(q_fwd_thr_gain) && is_positive(q_fwd_tilt_lim)) {
+        // Handle the case where we are limiting the forward pitch angle to prevent negative wing lift
+        // and are using the forward thrust motor or tilting rotors to provide the forward acceleration
+        float fwd_tilt_rad = radians(constrain_float(-0.01f * (float)plane.nav_pitch_cd, 0.0f, 45.0f));
+        q_fwd_nav_pitch_lim_cd = (int32_t)MIN(-100.0f * q_fwd_tilt_lim, 0.0f);
+        plane.nav_pitch_cd = MAX(plane.nav_pitch_cd, q_fwd_nav_pitch_lim_cd);
+        q_fwd_throttle = MIN(q_fwd_thr_gain * tanf(fwd_tilt_rad), 1.0f);
+
+        // To prevent forward motor prop strike, reduce throttle to zero when close to ground
+        // When we are doing horizontal positioning in a VTOL land
+        // we always allow the fwd motor to run. Otherwise a bad
+        // lidar could cause the aircraft not to be able to
+        // approach the landing point when landing below the takeoff point
+        if (!in_vtol_land_approach()) {
+            float alt_cutoff = MAX(0,vel_forward_alt_cutoff);
+            float height_above_ground = plane.relative_ground_altitude(plane.g.rangefinder_landing);
+            float fwd_thr_scaler = linear_interpolate(0.0f, 1.0f, height_above_ground, alt_cutoff, alt_cutoff+2);
+            q_fwd_throttle *= fwd_thr_scaler;
```

Comment:
This leaves the vehicle with no forward control. Forward throttle set to 0 and pitch angle is still constrained.

Reply:
Fixed. I added line below to fade the forward pitch limit to Q_ANGLE max as forward throttle is faded to zero.